### DISCUSSION TITLE
Whether filters such as HSLColorFilter are outdated

### DISCUSSION THREAD

**Discussion by luohuiiqng** [pelicanmapping/osgearth#2406] [Support Q&A] [answered]
Hello, I am currently trying to find some solutions to fine-tune the appearance of the layers. Through queries I learned that filters such as HSLColorFilte can do this. However, although these classes exist in the source code, the library built by CMake does not contain them. So are these classes obsolete, or is it the way I built them? Thank you!
![微信图片_20231211084935](https://github.com/gwaldron/osgearth/assets/82392471/c7c07120-31dc-4c79-8f42-2e51f04f438a)

**Comment by gwaldron** [comment, marked as answer]
The "color filter" approach is indeed deprecated. The modern approach is the use layer shaders. Examples include `glsl.earth` or XML include files like `shader_adjust_hsl.xml` in the tests folder

**Reply by luohuiiqng** [reply]
Thank you!
